### PR TITLE
Added `ChatMode` property to the `LogOnDetails` class

### PR DESCRIPTION
**Pull request by Rei-sen**
This lets the user enable new Steam group chat

### CODE REVIEW

File: SteamKit2/SteamKit2/Steam/Handlers/SteamUser/SteamUser.cs [hunk 1]
```diff
@@ -111,6 +111,12 @@ public sealed class LogOnDetails
             /// </summary>
             /// <value>The machine name.</value>
             public string? MachineName { get; set; } = $"{Environment.MachineName} (SteamKit2)";
+            /// <summary>
+            /// Gets or sets the chat mode.
+            /// Setting this field to 2 enables support for new Steam group chat.
```
Comment (yaakov-h) [inline SteamKit2/SteamKit2/Steam/Handlers/SteamUser/SteamUser.cs:116]: This sounds more like an enum than a uint.

**Review by JustArchi — APPROVED** [review]

**Comment by JustArchi** [comment]
Looks much cleaner than announcing chat mode after login, which is what I'm doing in ASF right now.

Any objections to merge this?

File: SteamKit2/SteamKit2/Steam/Handlers/SteamUser/SteamUser.cs (hunk 1 above)
Comment (xPaw) [reply SteamKit2/SteamKit2/Steam/Handlers/SteamUser/SteamUser.cs:132]: You can remove this string now that it's actually an enum.

File: SteamKit2/SteamKit2/Steam/Handlers/SteamUser/SteamUser.cs [hunk 2]
```diff
@@ -253,6 +275,8 @@ public void LogOn( LogOnDetails details )
             logon.Body.machine_name = details.MachineName;
             logon.Body.machine_id = HardwareUtils.GetMachineID( Client.Configuration.MachineInfoProvider );
 
+            logon.Body.chat_mode = ( uint )details.ChatMode;
```
Comment (xPaw) [inline SteamKit2/SteamKit2/Steam/Handlers/SteamUser/SteamUser.cs:278]: Perhaps wrap this into an if chatmode is not default?

**Pull request closed without merging**